File: backend/src/test/java/com/festago/support/fixture/TicketFixture.java
package com.festago.support.fixture;

import com.festago.stage.domain.Stage;
import com.festago.ticket.domain.Ticket;
import com.festago.ticket.domain.TicketType;

@Deprecated
public class TicketFixture {

    private Long id;
    private Stage stage = StageFixture.builder().build();
    private TicketType ticketType = TicketType.VISITOR;
    private Long schoolId = 1L;

    private TicketFixture() {
    }

    public static TicketFixture builder() {
        return new TicketFixture();
    }

    public TicketFixture id(Long id) {
        this.id = id;
        return this;
    }

    public TicketFixture stage(Stage stage) {
        this.stage = stage;
        return this;
    }

    public TicketFixture ticketType(TicketType ticketType) {
        this.ticketType = ticketType;
        return this;
    }

    public TicketFixture schoolId(Long schoolId) {
        this.schoolId = schoolId;
        return this;
    }

    public Ticket build() {
        return new Ticket(id, stage, ticketType, schoolId);
    }
}

File: android/festago/app/src/test/java/com/festago/festago/presentation/fixture/TicketFixture.kt
package com.festago.festago.presentation.fixture

import com.festago.festago.domain.model.MemberTicketFestival
import com.festago.festago.domain.model.Stage
import com.festago.festago.domain.model.Ticket
import com.festago.festago.domain.model.TicketState
import java.time.LocalDateTime

object TicketFixture {
    fun getMemberTicket(ticketId: Long = 0): Ticket {
        return Ticket(
            id = ticketId,
            number = 0,
            entryTime = LocalDateTime.MIN,
            state = TicketState.AWAY,
            reserveAt = LocalDateTime.MAX,
            stage = getStage(),
            festivalTicket = getMemberTicketFestival(),
        )
    }

    fun getStage(stageId: Int = 0) = Stage(id = stageId, startTime = LocalDateTime.MIN)

    fun getMemberTicketFestival(festivalId: Int = 0) = MemberTicketFestival(
        id = festivalId,
        name = "test",
        thumbnail = "test.com",
    )

    fun getMemberTickets(ids: List<Long>) = ids.map { id ->
        getMemberTicket(id)
    }
}
